bugsnag-plugin-android-ndk/src/main/assets/include/event.h

```diff
@@ -163,6 +170,20 @@ bool bugsnag_event_is_unhandled(void *event_ptr);
 char *bugsnag_event_get_grouping_hash(void *event_ptr);
 void bugsnag_event_set_grouping_hash(void *event_ptr, char *value);
 
+/* Metadata */
+
+void bugsnag_event_add_metadata_double(void *event_ptr, char *section, char *name, double value);
+void bugsnag_event_add_metadata_string(void *event_ptr, char *section, char *name, char *value);
+void bugsnag_event_add_metadata_bool(void *event_ptr, char *section, char *name, bool value);
+
+void bugsnag_event_clear_metadata_section(void *event_ptr, char *section);
+void bugsnag_event_clear_metadata(void *event_ptr, char *section, char *name);
+
+bsg_metadata_t bugsnag_event_has_metadata(void *event_ptr, char *section, char *name);
+double bugsnag_event_get_metadata_double(void *event_ptr, char *section, char *name);
+void bugsnag_event_get_metadata_string(void *event_ptr, char *section, char *name, char dst[64]);
```

Comment:
All the other `get` methods for strings return a `char *` to the actual string in the event structure. Is there a reason we can't take this approach for metadata strings rather than passing in a write parameter?

Reply:
C doesn't allow returning arrays from functions